demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import Iterable, List
+
+from demisto_sdk.commands.common.constants import GitStatuses
+from demisto_sdk.commands.content_graph.objects.dashboard import Dashboard
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Dashboard
+
+FIELDS_TO_INCLUDE = ["fromDate", "toDate", "fromDateLicense"]
+
+
+class IsDashboardContainNecessaryFieldsValidator(BaseValidator[ContentTypes]):
+    expected_git_statuses = [GitStatuses.ADDED]
+    error_code = "DA101"
+    description = "Validate that the dashboard includes all the necessary fields."
+    rationale = "The Dashboard should contains the required 'fromDate', 'toDate', 'fromDateLicense' fields."
+    dashboard_error_message = (
+        "the '{0}' fields are missing from {1} and need to be added."
+    )
+    widget_error_message = (
+        "the '{0}' fields are missing from {1} Widget in {2} and need to be added."
+    )
+    related_field = ""
+    is_auto_fixable = False
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        results = []
+        error_messages = []
+        for content_item in content_items:
+            dashboard_missing_fields = self.dashboard_missing_necessary_fields(
+                content_item.data
+            )
+            widget_missing_fields = self.widget_missing_necessary_fields(
+                content_item.data
+            )
```

Comment:
We shouldn't work with the content item data.
We're using graph objects here so the idea is to use the object fields - if necessary then you can add that to the graph object.